babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -539,4 +539,17 @@ private void checkParseInfixCast(String sqlType) {
     final String expected = "SELECT MOD(`FOO`, `BAR`)";
     sql(sql).ok(expected);
   }
+
+  @Test public void testMergeIntoBigQuery() {
```

Comment:
We should add a test verifying that the old unparsing logic still works on the same example query when not in BigQuery SQL dialect.

Reply:
I'll add the test. About the main comment: the only way that I can think of for now is putting if statements into the original file that checks the dialect. For this specific change that might actually be alright as it is quite a simple difference between the two. I can also spend a bit more time trying to explore different solutions.

Reply:
One idea: can we leave the merge unparsing logic where it is today, but move the insert logic only into the dialect files?

Reply:
Do you mean something like writer.getDialect().unparseInsertStatement(..) in which the default logic will be as before and BigQuery specific will be my change?

Reply:
Yes, exactly.